nested-labvm/atd-docker/uilanding/src/uilanding.py

```diff
@@ -282,6 +305,52 @@ def pS(mtype):
     mmes = "\t" + mtype
     print("[{0}] {1}".format(cur_dt, mmes.expandtabs(7 - len(cur_dt))))
 
+class LabHandler(tornado.web.RequestHandler):
+    def get(self):
+        self.set_header("Access-Control-Allow-Origin", "*")
+        selected_lab_option = self.get_argument('lab_value')
+        docker_conn= docker.from_env()
+        login_container = docker_conn.containers.get('atd-login')
+        container_output=login_container.exec_run(f'python3 /usr/local/bin/callConfigTopo.py  {DEFAULT_MENU_FILE_VALUE} {selected_lab_option}')
+        print(container_output)
+        log_file = open('log.txt','w')
+        log_file.write(str(container_output.output.decode("utf-8")))
+        log_file.close()
+        with open("log.txt", "r") as txt_file:
+            response =  txt_file.readlines()
+        self.write({
+            'response':response
+        })
+
+class LabStausHandler(tornado.web.RequestHandler):
+    def get(self):
+        self.set_header("Access-Control-Allow-Origin", "*")
+        docker_conn= docker.from_env()
+        login_container = docker_conn.containers.get('atd-login')
+        container_output=login_container.exec_run(f'sudo lab_status.py')
+        log_file = open('log.txt','w')
+        log_file.write(str(container_output.output.decode("utf-8")))
+        log_file.close()
+        with open("log.txt", "r") as txt_file:
+            response =  txt_file.readlines()
+        print(response)
+        self.write({
+            'response':response
```

Comment:
also account for the scenario:
if the container exits with an error, then the error is logged in the log.txt file. in such case, some default error message should be sent as response.

Reply:
That is fine, 99% atd-login container will not fail because it is must required container.
switch details are extracted from files which are created as part of build process.
 lab_status.py have try/catch functionality, switch will be down if connectivity is failed. 
collectively there are very  less chances to occur error